src/common/vector_operations/vector_cast.cpp

```diff
@@ -132,6 +132,10 @@ static void string_cast_switch(Vector &source, Vector &result, SQLType source_ty
 		UnaryExecutor::Execute<string_t, timestamp_t, duckdb::CastToTimestamp, true>(source, result, count);
 		break;
 	default:
+		if(source_type == SQLType::BLOB && target_type == SQLType::VARCHAR) {
```

Comment:
This should be integrated into the actual switches. There needs to be a new set of casts for BLOB. I think the only possible cast for the blob type is the BLOB -> STRING cast.

The cast from BLOB to STRING also needs to do conversion. SQLType::VARCHAR has to be valid UTF8, and blobs can contain arbitrary bytes. The conversion I am thinking of is the conversion to hexadecimal, similar to what should be done in Value::ToString() for the blob type.

Reply:
I also think that the only cast needs is BLOB to STRING and STRING to BLOB, I will take a better look at it. 
The other casts involving numeric are not valid in Postgres such as:
select 1::bytea;
select 1.0::bytea;